pkg/controller/kieapp/defaults/merge_test.go

```diff
@@ -289,7 +289,6 @@ func TestMergeBuildConfigandIStreams(t *testing.T) {
 	assert.Equal(t, "test-kieserver", server.BuildConfigs[0].ObjectMeta.Name)
 	assert.Empty(t, server.DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Namespace)
 	assert.Equal(t, "test-kieserver:latest", server.DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Name)
-	assert.Equal(t, "test-kieserver", server.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Image)
```

Comment:
instead of removing this, is there a different check we should be running instead? also, do we have coverage for these immutable envs when a build is NOT configured?

Reply:
Since without build we won't have bc or ImageStream, so I don't think there is any alternative for this particular test, but we add some unit test for the production immutable env, I added the image verification as part of that change.

Reply:
the kie server has to use an image, otherwise we have a failed deployment.. or a DC just sitting there. does an immutable env not have a default image when BC not used?

Reply:
That's what I defined in the test. The default image for kieserver when BC is not defined
It should be registry.redhat.io/rhpam-7/rhpam-kieserver. Does it need to be something else?